auto_create.go

```diff
@@ -1026,3 +1036,72 @@ func deleteOrphanedAutoCreatedApplications(resController *ClusterWatcher) error
 
 	return nil
 }
+
+
+/* containsName check if an array contains name string */
+func containsName(array []string, name string) bool {
+	if array != nil {
+		for _, a := range array {
+			if a == name {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+/* updateAutoCreateAppNameMap update autoCreateAppNameMap when an app is added or deleted */
+func updateAutoCreateAppNameMap(resourceInfo *autoCreateResourceInfo, operation string) {
+	if logger.IsEnabled(LogTypeEntry) {
+		logger.Log(CallerName(), LogTypeEntry, "updateAutoCreateAppNameMap operation: " + operation)
+	}
+	if resourceInfo != nil {
+		nameObj := resourceInfo.metadata[NAME]
+		if nameObj != nil {
+			name := nameObj.(string)
+			if logger.IsEnabled(LogTypeEntry) {
+				logger.Log(CallerName(), LogTypeEntry, "app name: " + name)
+				logger.Log(CallerName(), LogTypeEntry, "autoCreateName: " + resourceInfo.autoCreateName)
+			}
+			values, ok := autoCreateAppNameMap[resourceInfo.autoCreateName]		
+			//an app is added or updated	
+			if operation == "add" {
+				if !ok {
+					if logger.IsEnabled(LogTypeDebug) {
+						logger.Log(CallerName(), LogTypeDebug, "autoCreateName does not exist - add it to map")				
+					}
+					autoCreateAppNameMap[resourceInfo.autoCreateName] = []string{name}
+				} else {									
+					//add app name to array if it does not exist
+					if !containsName(values, name) {
+						if logger.IsEnabled(LogTypeDebug) {
+							logger.Log(CallerName(), LogTypeDebug, "autoCreateName exists - add app name to array: " + name)				
+						}
+						values = append(values, name)
+						autoCreateAppNameMap[resourceInfo.autoCreateName] = values
+					}
+				}
+			} else { //delete operation				
+				//an app is deleted 		
+				for i, fromName := range values {
+					if logger.IsEnabled(LogTypeDebug) {
+						logger.Log(CallerName(), LogTypeDebug, "delete app from array " + name)				
+					}
+					if name == fromName {
+						values = append(values[:i], values[i+1:]...)
+						autoCreateAppNameMap[resourceInfo.autoCreateName] = values
+						break
```

Comment:
This looks like it will break before all values (except the one being deleted) have been appended to "values".  Will this loop lose names ?  Also on line 1091,  values appends to itself.  But it also is used by the range iterator on line 1086.  That seems wrong.  Shouldn't lines 1091 and 1092 use a different variable for the updated list ?

Reply:
It will reset values after it is appended and it will not lose names.